optimum/exporters/onnx/base.py

```diff
@@ -596,7 +596,8 @@ def generate_dummy_inputs(self, framework: str = "pt", **kwargs):
             and self.use_cache_branch is not False
             and "attention_mask" in dummy_inputs
         ):
-            past_length = dummy_inputs["past_key_values"][0][0].shape[2]
+            past_length = dummy_inputs["past_key_values"][0][0].shape[-2]
```

Comment:
@fxmarty here I adapt the indexing for fetching the past seq len, will it impact bloom?

Reply:
ok, it's due to the fact that for bloom the shape of key and value are transposed in the last two dims.